### PR TITLE
fix: Prevent duplicate token when renaming

### PR DESCRIPTION
**Pull request by kof**
Fixes https://github.com/webstudio-is/webstudio/issues/3776

## Description

During renaming of the token

- when token name is less than 1 character - show error state
- when token name already exists - show error state

Fixed a bunch of bugs with esc, error handling and switching to editing another token while still editing the first one

https://github.com/user-attachments/assets/5833e611-235b-4c2c-a3ed-3ee521ac5fd1

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

File: apps/builder/app/builder/features/style-panel/style-source-section.tsx [hunk 1]
```diff
@@ -363,11 +367,27 @@ const reorderStyleSources = (styleSourceIds: StyleSource["id"][]) => {
...
+  if (name.trim().length === 0) {
+    return { name: "minlength", id };
+  }
+  for (const styleSource of styleSources.values()) {
+    if (
+      styleSource.type === "token" &&
+      styleSource.name === name &&
+      styleSource.id !== id
+    ) {
+      return { name: "duplicate", id };
```
Comment (TrySound) [inline apps/builder/app/builder/features/style-panel/style-source-section.tsx:384]: "type"? "name" got me confused.

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/builder/features/style-panel/style-source-section.tsx (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/style-panel/style-source-section.tsx:384]: i thought its like error.name in new Error

Comment (kof) [reply apps/builder/app/builder/features/style-panel/style-source-section.tsx]: renamed

**Comment by TrySound** [comment]
image

**Comment by kof** [comment]
@TrySound I don't see this layout broken neither in the branch nor on main, I think something is broken on your instance

**Comment by kof** [comment]
Reproduced it after restarting dev server

**Review by TrySound — APPROVED** [review]

**Pull request merged**